dmscripts/publish_draft_services.py

```diff
@@ -188,9 +192,13 @@ def get_draft_services_iter(client, framework_slug, draft_ids_file=None):
 
     else:
         for sf in supplier_frameworks:
-            yield from client.find_draft_services_by_framework_iter(
-                framework_slug, status='submitted', supplier_id=sf['supplierId']
-            )
+            try:
+                yield from client.find_draft_services_by_framework_iter(
+                    framework_slug, status='submitted', supplier_id=sf['supplierId']
+                )
+            except StopIteration:
+                # Handle exception for Python 3.7 and above
+                return None
```

Comment:
This is quite weird, looking at [`make_iter_method`](https://github.com/alphagov/digitalmarketplace-apiclient/blob/2158a624cf52d49ae8596e84fef03e60bce87ae0/dmapiclient/base.py#L25) in digitalmarketplace-apiclient, we shouldn't be raising `StopIteration` in `find_draft_services_by_framework_iter`...

Reply:
I think it's because we're `yield from`-ing.

Reply:
I think it's the mock that is causing the error, rather than production code.

We could either leave this `try... except` block inplace